substrate/bin/utils/chain-spec-builder/bin/main.rs

```diff
@@ -65,8 +69,29 @@ fn inner_main() -> Result<(), String> {
 				.map_err(|e| format!("to pretty failed: {e}"))?;
 			fs::write(chain_spec_path, chain_spec_json).map_err(|err| err.to_string())?;
 		},
+		ChainSpecBuilderCmd::AddCodeSubstitute(AddCodeSubstituteCmd {
+			ref input_chain_spec,
+			ref runtime_wasm_path,
+			block_height,
+		}) => {
+			let chain_spec = ChainSpec::from_json_file(input_chain_spec.clone())?;
+
+			let mut chain_spec_json =
+				serde_json::from_str::<serde_json::Value>(&chain_spec.as_json(false)?)
+					.map_err(|e| format!("Conversion to json failed: {e}"))?;
+
+			set_code_substitute_in_json_chain_spec(
+				&mut chain_spec_json,
+				&fs::read(runtime_wasm_path.as_path())
+					.map_err(|e| format!("Wasm blob file could not be read: {e}"))?[..],
+				block_height,
+			);
+			let chain_spec_json = serde_json::to_string_pretty(&chain_spec_json)
+				.map_err(|e| format!("to pretty failed: {e}"))?;
+			fs::write(chain_spec_path, chain_spec_json).map_err(|err| err.to_string())?;
+		},
```

Comment:
nit: to avoid code duplication in `UpdateCode` and `AddCodeSubstitute` we could add a function `update_chain_spec_file_in_place` which could take a closure that modifies `&mut chain_spec_json` provided as its input.